app/models/validations/household_validations.rb

```diff
@@ -73,6 +73,15 @@ def validate_previous_housing_situation(record)
     end
   end
 
+  def validate_referral(record)
+    if record.rsnvac == "Relet to tenant who occupied same property as temporary accommodation" && REFERRAL_INVALID_TMP.include?(record.referral)
+      record.errors.add :referral, I18n.t("validations.household.referral.rsnvac_non_temp")
+    end
+    if record.tenancy == "Secure (including flexible)" && record.referral.present? && record.referral != "Internal transfer"
+      record.errors.add :referral, I18n.t("validations.household.referral.secure_tenancy")
```

Comment:
Should this add the error to `rsnvac` as well? i.e. could you answer them in either order?

Reply:
Hmm, so this is a validation that I just moved. Seems like there's a validation for it on line 70 of the same file: `validate_previous_housing_situation`, do you remember if they will run on both pages if they're in one method?

Reply:
Yeah all validations run on all pages (on all saves) so shouldn't need to duplicate any.

Reply:
I was wrong about the `validate_previous_housing_situation` validation, it's actually checking a different field 🤦 
I located the correct one in the property validation tho and moved the referral validation there

Reply:
So I think you can technically answer `referral` first and then `rsnvac` right? In which case we'd need to show the error message on the `rsnvac` page?

Reply:
Yup! That's happening